[nicljr] iP

nicljr opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from here
2. double-click it
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

nicljr committed:
no message

nicljr committed:
Duke Level 2: Add, List

nicljr committed:
Duke Level 3 Mark As Done

nicljr committed:
Duke Level 4: ToDo, Deadline, Event

nicljr committed:
Duke: A-TextUiTesting

nicljr committed:
Duke Level-5: Handle Errors

nicljr committed:
Duke Level 6: Delete

nicljr committed:
Duke Level-7: Save and Load

nicljr committed:
Duke Level 8: Dates and Times

nicljr committed:
Duke A-MoreOOP: Use More OOP

nicljr committed:
no message

nicljr committed:
no message

nicljr committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nicljr committed:
Duke: A-Packages

nicljr committed:
Updated Commands and TaskList

nicljr committed:
Fixed Dates and Times

harin0826 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.Scanner;
+
+public class Ui {
+    public static String HELLO = "Hello! I am Duke Nice To Meet You\n";
```
Could HELLO_MSG be more explanatory for this variable?

harin0826 reviewed (commented):
Learnt a lot from your code! Just some suggestions to improve code quality. Thank you!! :)

praveenkrishna0512 commented on src/main/java/duke/TaskAssigner.java:
```diff
@@ -0,0 +1,82 @@
+
+    public TaskAssigner() {}
+
+    public Task assignTask(String command) throws DukeException {
```
I like that your function name is written beginning with a verb! It helps to convey the action which the function is performing.

praveenkrishna0512 commented on src/main/java/duke/TaskAssigner.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    public Task assignEvent(String command) throws DukeException {
+        String[] splitTimings = command.split("/from | /to ");
```
May I clarify the intention behind the naming of this variable? Perhaps another name could be choosen to convey the message that the variable is an array of strings representing time (ie. `timestampsAsString`)

praveenkrishna0512 commented on src/main/java/duke/TaskAssigner.java:
```diff
@@ -0,0 +1,82 @@
+                    "event {desc} /from yyyy-mm-dd hhmm /to yyyy-mm-dd hhmm\n");
+        }
+        try {
+            int s_index = command.indexOf("/from") + 6;
+            int e_index = command.indexOf("/to") + 4;
+            String start_date = TimeChecker.updateTime(command.substring(s_index, e_index - 5));
+            String end_date = TimeChecker.updateTime(command.substring(e_index));
+            String e_desc = command.substring(6, s_index - 6);
```
Not sure if I like the naming of these variables, as they do not follow `camelCase`.

praveenkrishna0512 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,63 @@
+        System.out.println(HELLO);
+    }
+
+    public void bye() {
```
May I clarify the intention behind the naming of this function? Perhaps you could consider `printBye()` instead, so as to convey the action performed by the function.

praveenkrishna0512 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+public class Deadline extends Task {
+
+    static protected String DEFAULT_TIME = "2359";
+    protected LocalDateTime by;
```
Not sure if others will understand what the `by` variable represents on first look! Perhaps you could consider renaming it to `dateBy` so as to convey to others that it is a date!

praveenkrishna0512 reviewed (commented):
Overall, the code structure looks good! There were a few comments regarding following proper coding standards for naming. However, it is still a very good job done :)

nicljr committed:
Duke A-JUnit: Testing behaviour of Code + Updated Printing Methods for Tasks

nicljr committed:
Duke: A-JavaDoc

nicljr committed:
Duke: A-CodingStandard

nicljr committed:
Duke Level-9: Find Command

jedidiahC commented on src/main/java/duke/TaskAssigner.java:
```diff
@@ -45,33 +45,38 @@ public Task assignToDo(String command) {
+            String updated_deadline = TimeChecker.updateTime(deadline);
             String d_desc = command.substring(9, d_index - 4);
-            return new Deadline(d_desc, deadline);
+            return new Deadline(d_desc, updated_deadline);
```
Might want to consider changing `updated_deadline` to `updatedDeadline` to adhere to the Java coding standard of naming variables in camelCase. I noticed that you used snake_case instead of camelCase in several other areas of your code as well. Might I suggest using checkStyle to detect such cases which could save you a lot of time instead of finding them manually.

jedidiahC commented on src/main/java/duke/Parser.java:
```diff
@@ -29,12 +34,24 @@ public static Command stringToCommand(String command) {
+        return localDateTime;
+    }
+
+    public static String dateTimeToString(LocalDateTime localDateTime) {
```
Should methods such as `deadlineSplit` be named something like `splitDeadlineIntoStringArray` to better follow the convention of naming methods using verbs?

jedidiahC commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,112 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
```
Should Parser have a descriptive header comment?

jedidiahC commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,93 @@
+public class Storage {
+
+    public static String path = "DukeData/tasks.txt";
+    public static String dirPath = "DukeData";
```
Should this variable be converted to a constant?

jedidiahC commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,93 @@
+            Task curr = new Task("");
+
+            switch (text[0]) {
+                case ("T"):
```
Should the indentation for case clauses be removed to follow the coding standard? 
```suggestion
            switch (text[0]) {
            case ("T"):
```

jedidiahC commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,121 @@
+import java.util.ArrayList;
+
+public class Ui {
+    public static String GREET_MSG = "Hello! I am Duke Nice To Meet You\n";
```
Nice use of uppercase with underscores for naming the constants!

jedidiahC reviewed (commented):
Overall, I think most of your code following the Java coding standard! 

But you may want to take note of the few instances where you used snake_case instead of camelCase for naming variables. These issues should be detected by using checkStyle. 

Otherwise, the layout and statements seem to follow the coding standard. Good work!

nicljr committed:
Duke Level 10: Completeing JavaFX Tutorial

nicljr committed:
Duke Level 10: GUI/ Updated Command and UI

nicljr committed:
Duke A-CheckStyle

nicljr committed:
Invalid events are accepted by Duke, where the end date is earlier than start date.

An event with an end date canonot be earlier than the start date, this is to ensure that Duke catches this error and doesn't accept this invalid input. We want to ensure the correctness of our Tasks in Duke.

Event constructor will throw a DukeException if such an input arrives, by utilising the LocalDateTime isBefore() method.

Event holds the LocalDateTime information, hence one way I thought was to throw the Exception within the Event Class.

nicljr committed:
Duke A-Assertions

nicljr committed:
Revert "Duke A-Assertions"

This reverts commit 35810a92141149eb303d351dd6e8b6bda97cc8c5.

nicljr committed:
Duke A-Assertions

nicljr committed:
Merge pull request #1 from nicljr/A-Assertions

Duke A-Assertions

nicljr committed:
Duke A-Jar: JAR File created for v0.2 upgraded from v0.1

nicljr committed:
Duke Level-9: Making Find more specific

nicljr committed:
Revert "Duke Level-9: Making Find more specific"

This reverts commit 00bdbb8606324178578be7b5ef64e67ef8be39cf.

nicljr committed:
Revert "Revert "Duke Level-9: Making Find more specific""

This reverts commit 3bceb3af0fc680105303bb1f2fe320d0bea8618f.

nicljr committed:
A-CodingStandard: Fix Coding Standards

Coding Standard needs to be adhered for clarity

Removed Magic Numbers
Ensured no Deep Nesteing, ArrowHead
No long method names
Ensure clarity of method names

Done as suggested by the textbook provided

nicljr committed:
Merge pull request #3 from nicljr/A-CodeQuality

A-CodingStandard: Fix Coding Standards

nicljr committed:
Duke C-Statistics: Provided Statistics on the added tasks within the past week and Completed Task within the past week. Provided Insight on User Productivity through checking Urgency of various deadline tasks.

nicljr committed:
Update README.md

nicljr committed:
Update README.md

nicljr committed:
JavaDocs and CheckStyle

nicljr committed:
Merge branch 'master' of https://github.com/nicljr/ip

nicljr committed:
Duke A-UserGuide

nicljr committed:
JAR-Release Updates
